red_blue_world/Quilt.py

```diff
@@ -13,13 +16,119 @@ class Quilt:
     def __init__(self):
         self._patches: Dict[PatchID, Patch] = {}
         self._active_patch_id: PatchID = (0, 0)
-        self._active_patch: Patch = ...
+        self._active_patch: Patch = self.build_patch()
```

Comment:
Technically `self._active_patch` and `self._active_patch_id` are redundant, and storing the same info twice increases bug surface area (i.e. we now how to ensure these are kept in sync). However, these are in the performance hot-path and storing both allows us to reduce an additional dictionary lookup on every tick. I _think_ the nearly free performance bump is worth the bug risk.